MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -65,7 +65,7 @@ def extract_jenkins_link_and_testname(content):
     jenkins_links = []
     test_names = []
 
-    pattern = r"(https:\/\/.+\/job\/Test_openjdk\d+.+\/\d+)"
+    pattern = r"(https:\/\/.+\/job\/Test_openjdk\d+.+\/\d+\/?(?=console))"
```

Comment:
Since most jenkins links do not have `console` key words, for your case that getting a long string with spaces inside, you can put this line https://github.com/adoptium/aqa-test-tools/blob/538b6905673eb771b39d822a5f4b69ef42f77ab5/MachineLearningPrototype/utils/preprocess_data.py#L78 before applying this pattern, and use `pattern`  to `splitted_content` instead.